util/gen_effect_id.ts

```diff
@@ -74,7 +74,7 @@ const makeEffectMap = (table: Table<'#', 'Name'>) => {
   const map = new Map<string, string>();
   for (const [id, effect] of Object.entries(table)) {
     const rawName = effect['Name'];
-    if (!rawName)
+    if (rawName === undefined || rawName.length === 0)
```

Comment:
The check on line 81 doesn't cause issues because it's not a nullable string (`cleanName` returns type `string`, not `string | undefined`).

I think that maybe we need a different eslint rule to cover this case?

Reply:
(this suggestion has been implemented but I'm leaving this unresolved to discuss the line 81 question)

Reply:
Are you suggesting that we have an eslint rule that says "if you want to check for an empty string you need to do it explicitly via `someString === ''` and not say `!someString`?

Reply:
You were asking why the check on line 81 wasn't causing issues, and it's because `name` can't be `undefined` or `null`. If you feel like it *should* be throwing a warning or error, that might be the `allowString` parameter here which defaults to `true`:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md#allowstring

Reply:
Ah, sorry sorry.  Once you explained it a few comments earlier, that made sense to me why it wasn't throwing.  Given the eslint rule that you're enabling now, it might be a good thing to follow up with.  I think it would certainly clarify these checks.

This is a weak "sounds good to enable this rule as a followup" from me, if you feel strongly that it's not a good idea, I'm happy to leave this as-is too.

Reply:
I can play around with enabling it after the rest of the changes are merged and see how big the impact is, I think that's probably the best option.